platform/lib/routers/growPages.js

```diff
@@ -168,6 +181,40 @@ function rewriteLinks(canonical, html, format, level) {
   return html;
 }
 
+/**
+ * Updates template context and URL:
+ *
+ * - matches the URL to the right template, based on the active format and version
+ * - adds the supported version to the template context
+ */
+function updateComponentInfo(request, context, url) {
+  // extract component and optional version from the path
+  const componentMatch = COMPONENT_NAME_PATTERN.exec(request.path);
+  if (!componentMatch) {
+    // nothing to do if it's not a component URL
+    return;
+  }
+  const component = componentMatch[2];
+  const version = componentMatch[4];
+  // get the supported version for this format
+  const versionsByFormat = formatComponentMapping[component];
+  if (!versionsByFormat) {
+    signale.warn(`No version mapping defined for ${component}. Run 'gulp importAll' to fix.`);
+    return;
+  }
+  // add format supported versions to template context
+  context.versions = versionsByFormat[context.format];
+  // rewrite path based on the format and available versions
+  const latestVersionByFormat = context.versions[context.versions.length - 1];
+  if (version === versionsByFormat.current) {
+    // change amp-carousel-v$LATEST => amp-carousel
+    url.pathname = url.pathname.replace(`-v${version}`, '');
+  } else if (versionsByFormat.current !== latestVersionByFormat) {
```

Comment:
Also check if the version is not set.
If a specific version was requested that is not the current version this code would add a second version part!